base_layer/wallet/src/wallet.rs

```diff
@@ -476,6 +477,27 @@ where
         }
     }
 
+    pub async fn get_wallet_address(&self) -> Result<TariAddress, WalletError> {
```

Comment:
```suggestion
    pub async fn get_wallet_dual_address(&self) -> Result<TariAddress, WalletError> {
```

Reply:
It can be single as well. 
This returns a type wallet address, not type dual address

Reply:
The point is it constructs `TariAddress::new_dual_address_with_default_features` only

Reply:
If you make it `get_wallet_dual_address`, it implies that the wallet has a `get_wallet_single_address` as well. 
We only want the address from the wallet, from the caller side it can be single as well, it does not matter. 

Reading this: `pub async fn get_wallet_dual_address(&self) -> Result<TariAddress, WalletError> {` I expect the return type to be `Result<DualAddress, WalletError>` which it is not.